android/capacitor/src/main/java/com/getcapacitor/CapConfig.java

```diff
@@ -1,147 +1,384 @@
+        webContentsDebuggingEnabled = (context.getApplicationInfo().flags & ApplicationInfo.FLAG_DEBUGGABLE) != 0;
+        webContentsDebuggingEnabled = JSONUtils.getBoolean(configJSON, "android.webContentsDebuggingEnabled", webContentsDebuggingEnabled);
+
+        // Plugins
+        pluginsConfiguration = new PluginsConfig(JSONUtils.getObject(configJSON, "plugins"));
     }
 
-    private JSONObject getConfigObjectDeepest(String key) throws JSONException {
-        // Split on periods
-        String[] parts = key.split("\\.");
+    public boolean isHTML5Mode() {
+        return html5mode;
+    }
 
-        JSONObject o = this.config;
-        // Search until the second to last part of the key
-        for (int i = 0; i < parts.length - 1; i++) {
-            String k = parts[i];
-            o = o.getJSONObject(k);
-        }
-        return o;
+    public String getServerUrl() {
+        return serverUrl;
     }
 
-    public String getString(String key) {
-        return getString(key, null);
+    public String getHostname() {
+        return hostname;
     }
 
-    public String getString(String key, String defaultValue) {
-        String k = getConfigKey(key);
-        try {
-            JSONObject o = this.getConfigObjectDeepest(key);
+    public String getAndroidScheme() {
+        return androidScheme;
+    }
 
-            String value = o.getString(k);
-            if (value == null) {
-                return defaultValue;
-            }
-            return value;
-        } catch (Exception ex) {}
-        return defaultValue;
+    public String[] getAllowNavigation() {
+        return allowNavigation;
     }
 
-    public boolean getBoolean(String key, boolean defaultValue) {
-        String k = getConfigKey(key);
-        try {
-            JSONObject o = this.getConfigObjectDeepest(key);
+    public String getOverriddenUserAgentString() {
+        return overriddenUserAgentString;
+    }
 
-            return o.getBoolean(k);
-        } catch (Exception ex) {}
-        return defaultValue;
+    public String getAppendedUserAgentString() {
+        return appendedUserAgentString;
     }
 
-    public int getInt(String key, int defaultValue) {
-        String k = getConfigKey(key);
-        try {
-            JSONObject o = this.getConfigObjectDeepest(key);
-            return o.getInt(k);
-        } catch (Exception ignore) {
-            // value was not found
-        }
-        return defaultValue;
+    public String getBackgroundColor() {
+        return backgroundColor;
     }
 
-    private String getConfigKey(String key) {
-        String[] parts = key.split("\\.");
-        if (parts.length > 0) {
-            return parts[parts.length - 1];
-        }
+    public boolean isMixedContentAllowed() {
+        return allowMixedContent;
+    }
+
+    public boolean isInputCaptured() {
+        return captureInput;
+    }
+
+    public Boolean getWebContentsDebuggingEnabled() {
```

Comment:
Is this supposed to be `Boolean`?

Also, if other boolean getters use `is`, should this as well?